tutorials/llm/llama-3/pruning-distillation/02_teacher_finetuning.ipynb

```diff
@@ -0,0 +1,153 @@
+    "cmd=\"torchrun --nproc-per-node=${TENSOR_PARALLEL_SIZE}\"\n",
+    "\n",
+    "${cmd} /opt/NeMo/examples/nlp/language_modeling/megatron_gpt_pretraining.py \\\n",
+    "    --config-path /opt/NeMo/examples/nlp/language_modeling/conf/ \\\n",
+    "    --config-name megatron_llama_distill.yaml \\\n",
+    "    \\\n",
+    "    name=${EXPERIMENT_NAME} \\\n",
+    "    \\\n",
+    "    exp_manager.exp_dir=${EXPERIMENT_DIR} \\\n",
+    "    exp_manager.checkpoint_callback_params.save_top_k=1 \\\n",
+    "    exp_manager.checkpoint_callback_params.save_nemo_on_train_end=True \\\n",
+    "    \\\n",
+    "    trainer.max_steps=${STEPS} \\\n",
+    "    trainer.log_every_n_steps=${LOG_INTERVAL} \\\n",
+    "    trainer.val_check_interval=${VAL_INTERVAL} \\\n",
+    "    trainer.limit_val_batches=${NUM_VAL_BATCHES} \\\n",
+    "    +trainer.num_sanity_val_steps=0 \\\n",
+    "    \\\n",
+    "    trainer.precision=bf16 \\\n",
+    "    trainer.devices=${TENSOR_PARALLEL_SIZE} \\\n",
+    "    trainer.num_nodes=${NODES} \\\n",
+    "    \\\n",
+    "    \"model.data.data_prefix={train:[1.0,$DATA_TRAIN],validation:[$DATA_VAL],test:[$DATA_TEST]}\" \\\n",
+    "    \\\n",
+    "    model.restore_from_path=${MODEL} \\\n",
+    "    +model.dist_ckpt_load_strictness=log_all \\\n",
+    "    \\\n",
+    "    ~model.tokenizer \\\n",
+    "    +model.tokenizer='{library: huggingface, type: meta-llama/Meta-Llama-3.1-8B, use_fast: True}' \\\n",
+    "    \\\n",
+    "    model.tensor_model_parallel_size=${TENSOR_PARALLEL_SIZE} \\\n",
+    "    model.sequence_parallel=True \\\n",
+    "    model.micro_batch_size=${MICRO_BATCH_SIZE} \\\n",
+    "    model.global_batch_size=${GLOBAL_BATCH_SIZE} \\\n",
+    "    \\\n",
+    "    model.encoder_seq_length=8192 \\\n",
+    "    model.num_layers=32 \\\n",
+    "    model.hidden_size=4096 \\\n",
+    "    model.ffn_hidden_size=14336 \\\n",
+    "    model.num_attention_heads=32 \\\n",
+    "    model.hidden_dropout=0.0 \\\n",
+    "    model.attention_dropout=0.0 \\\n",
+    "    model.apply_query_key_layer_scaling=True \\\n",
+    "    model.normalization='rmsnorm' \\\n",
+    "    model.bias=False \\\n",
+    "    model.activation='fast-swiglu' \\\n",
+    "    model.position_embedding_type='rope' \\\n",
+    "    model.share_embeddings_and_output_weights=False \\\n",
+    "    model.num_query_groups=8 \\\n",
+    "    ++model.scale_positional_embedding=True \\\n",
+    "    ++model.rotary_base=500000.0 \\\n",
+    "    \\\n",
+    "    model.optim.name=distributed_fused_adam \\\n",
+    "    model.optim.lr=${LR} \\\n",
+    "    model.optim.sched.min_lr=${MIN_LR} \\\n",
+    "    model.optim.sched.warmup_steps=${WARMUP_STEPS}"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "3040a993-8423-475f-8bc6-d1dd1ce16a83",
+   "metadata": {},
+   "source": [
+    "This will create a finetuned teacher model named `megatron_llama_ft.nemo` in `./distill_trainings/megatron_llama_ft/checkpoints/`. We'll use this later.\n",
```

Comment:
fix punctuation

 "This will create a fine-tuned teacher model named `megatron_llama_ft.nemo` in `./distill_trainings/megatron_llama_ft/checkpoints/`. We'll use this later.\n",